### PR TITLE
[BEAM-4419] Update Operator Testkit

### PR DESCRIPTION
**Pull request by VaclavPlajt**
Testkit was moved into 'euphoria-beam' package. Testing instrumentation was simplified.

DESCRIPTION HERE

------------------------

Follow this checklist to help us incorporate your contribution quickly and easily:

 - [X] Make sure there is a [JIRA issue](https://issues.apache.org/jira/projects/BEAM/issues/) filed for the change (usually before you start working on it).  Trivial changes like typos do not require a JIRA issue.  Your pull request should address just this issue, without pulling in other changes.
 - [X] Format the pull request title like `[BEAM-XXX] Fixes bug in ApproximateQuantiles`, where you replace `BEAM-XXX` with the appropriate JIRA issue.
 - [X] Write a pull request description that is detailed enough to understand:
   - [X] What the pull request does
   - [X] Why it does it
   - [X] How it does it
   - [ ] Why this approach
 - [X] Each commit in the pull request should have a meaningful subject line and body.
 - [X] Run `./gradlew build` to make sure basic checks pass. A more thorough check will be performed on your pull request automatically.
 - [ ] If this contribution is large, please file an Apache [Individual Contributor License Agreement](https://www.apache.org/licenses/icla.pdf).

### CODE REVIEW

File: sdks/java/extensions/euphoria/euphoria-beam/src/test/java/org/apache/beam/sdk/extensions/euphoria/beam/OperatorsTestSuite.java [hunk 1]
```diff
@@ -15,23 +15,48 @@
  * See the License for the specific language governing permissions and
  * limitations under the License.
  */
-package org.apache.beam.sdk.extensions.euphoria.operator.test;
+package org.apache.beam.sdk.extensions.euphoria.beam;
```
Comment (je-ik) [inline sdks/java/extensions/euphoria/euphoria-beam/src/test/java/org/apache/beam/sdk/extensions/euphoria/beam/OperatorsTestSuite.java:18]: This package looks a little odd to me. There is twice `beam`, could we find a little more descriptive subpackage name? Or maybe just drop the last `beam`?

**Review by je-ik — COMMENTED** [review]
I'd vote for changing the package name so that it is less redundant. I'm sure we can find something appropriate. :-)

File: sdks/java/extensions/euphoria/euphoria-beam/src/test/java/org/apache/beam/sdk/extensions/euphoria/beam/OperatorsTestSuite.java (hunk 1 above)
Comment (VaclavPlajt) [reply sdks/java/extensions/euphoria/euphoria-beam/src/test/java/org/apache/beam/sdk/extensions/euphoria/beam/OperatorsTestSuite.java:18]: I agree, the package name deserves better name. There is more cleanup work we need to do. So I created new [BEAM-4428](https://issues.apache.org/jira/browse/BEAM-4428) to serve as cleanup task TODO list.

**Review by mareksimunek — APPROVED** [review]
LGTM

File: sdks/java/extensions/euphoria/euphoria-beam/src/test/java/org/apache/beam/sdk/extensions/euphoria/beam/OperatorsTestSuite.java (hunk 1 above)
Comment (dmvk) [reply sdks/java/extensions/euphoria/euphoria-beam/src/test/java/org/apache/beam/sdk/extensions/euphoria/beam/OperatorsTestSuite.java:18]: 👍

File: sdks/java/extensions/euphoria/euphoria-beam/src/test/java/org/apache/beam/sdk/extensions/euphoria/beam/OperatorsTestSuite.java [hunk 2]
```diff
@@ -15,23 +15,48 @@
...
+    //    DistinctTest.class,
+    //    FilterTest.class,
+    FlatMapTest.class,
+    JoinTest.class,
+    //    JoinWindowEnforcementTest.class,
+    //    MapElementsTest.class,
+    ReduceByKeyTest.class,
+    //    ReduceStateByKeyTest.class,
+    //    SumByKeyTest.class,
+    //    TopPerKeyTest.class,
```
Comment (dmvk) [inline sdks/java/extensions/euphoria/euphoria-beam/src/test/java/org/apache/beam/sdk/extensions/euphoria/beam/OperatorsTestSuite.java:50]: can we try to uncomment more tests? I'd say that MapElementsTest, DistinctTest, CountByKeyTest, SumByKeyTest, TopPerKeyTest should work.

Comment (dmvk) [reply sdks/java/extensions/euphoria/euphoria-beam/src/test/java/org/apache/beam/sdk/extensions/euphoria/beam/OperatorsTestSuite.java:50]: and FilterTest

**Review by dmvk — APPROVED** [review]
LGTM

File: sdks/java/extensions/euphoria/euphoria-beam/src/test/java/org/apache/beam/sdk/extensions/euphoria/beam/OperatorsTestSuite.java (hunk 2 above)
Comment (VaclavPlajt) [reply sdks/java/extensions/euphoria/euphoria-beam/src/test/java/org/apache/beam/sdk/extensions/euphoria/beam/OperatorsTestSuite.java:50]: I will definitely try to do that. I believe that there will be some changes necessary to make these work, recalling the hassle we had last time with the operator testkit. I will see how it goes. New issue will be created if needed.

**Pull request merged**